runtime/src/chpl-file-utils.c

```diff
@@ -204,3 +203,58 @@ qioerr chpl_fs_remove(const char* name) {
     err = qio_mkerror_errno();
   return err;
 }
+
+qioerr chpl_fs_samefile(int* ret, qio_file_t* file1, qio_file_t* file2) {
+  qioerr err = 0;
+  struct stat f1;
+  struct stat f2;
+
+  int exitStatus = fstat(file1->fd, &f1);
+  if (exitStatus) {
+    // An error occurred.  Return it.
+    err = qio_mkerror_errno();
+    return err;
+  }
+  exitStatus = fstat(file2->fd, &f2);
+  if (exitStatus) {
+    // An error occurred.  Return it.
+    err = qio_mkerror_errno();
+  } else {
+    if (f1.st_dev == f2.st_dev && f1.st_ino == f2.st_ino) {
```

Comment:
It has been asked about, but I think (and I believe Brad has said so as well, though he may correct me) that stat is not something we want to expose.  The important parts of its functionality should be visible through functions like isFile, isDir, exists, and sameFile, but there's enough within it that we don't want which would be expected to be available by providing a pure Chapel stat().